From 9d231b8589b4a6ed5d9ff3744410e985e2ff5ddf Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Thu, 16 Jan 2020 22:58:24 +0200
Subject: [PATCH 01/16] net: mscc: ocelot: avoid incorrect consuming in skbs
 list

[ Upstream commit fc62c0948986b7aa6b2871a450a4469e35c5f9bc ]

Break the matching loop when find the matching skb for TX timestamp.
This is to avoid consuming more skbs incorrectly. The timestamp ID
is from 0 to 3 while the FIFO could support 128 timestamps at most.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Vladimir Oltean <vladimir.oltean@nxp.com>
---
 drivers/net/ethernet/mscc/felix_board.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/mscc/felix_board.c b/drivers/net/ethernet/mscc/felix_board.c
index 6c8684094381..a89d2b26ba13 100644
--- a/drivers/net/ethernet/mscc/felix_board.c
+++ b/drivers/net/ethernet/mscc/felix_board.c
@@ -497,6 +497,7 @@ static void felix_tx_clean(struct ocelot *ocelot)
 
 			list_del(pos);
 			devm_kfree(ocelot->dev, entry);
+			break;
 		}
 
 		if (likely(skb)) {
-- 
2.17.1

